### PR TITLE
Start replacing ACCEPTS table by parsing numpydoc.

### PR DESCRIPTION
**Pull request by anntzer**
In most cases, we can just look for 'param : type' in the docstring.

We let ACCEPTS still have the priority to handle weird cases such as
set_xlim where we document the first arg as "bottom", but it's "bottom,
top" for the purpose of the ACCEPTS table; resolving *that* problem is
punted to later.

Only removed a few ACCEPTS entries as a proof of principle.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by anntzer** [comment]
Well I did need to change artist.py to update the regex matching... (I didn't need to update the docstring, though, indeed.)

File: lib/matplotlib/axes/_base.py [hunk 1]
```diff
@@ -3268,11 +3242,9 @@ def set_xticklabels(self, labels, fontdict=None, minor=False, **kwargs):
         """
         Set the x-tick labels with list of string labels.
 
-        .. ACCEPTS: list of string labels
-
         Parameters
         ----------
-        labels : list of str
+        labels : List[str]
```
Comment (timhoffm) [inline lib/matplotlib/axes/_base.py]: Did we somewhere agree to switch to type annotation syntax?

Comment (anntzer) [reply lib/matplotlib/axes/_base.py]: Grepping for 'List\[' shows that it's definitely already being used.
```
lib/matplotlib/contour.py
576:        lw = self.get_label_width(self.labelLevelList[lmin],
577:                                  self.labelFmt, self.labelFontSizeList[lmin])
591:        self.add_label(xmin, ymin, rotation, self.labelLevelList[lmin],
592:                       self.labelCValueList[lmin])
1039:        artists : List[`.Artist`]
1042:        labels : List[str]
1810:        hatches : List[str], optional

lib/matplotlib/collections.py
386:        urls : List[str] or None
387:            .. ACCEPTS: List[str] or None

lib/matplotlib/tight_layout.py
52:    num1num2_list : List[int]

lib/matplotlib/widgets.py
515:        labels : List[str]
518:        actives : List[bool], optional
2484:    button : List[Int], optional

lib/matplotlib/backend_bases.py
1257:    callbacks : List[Tuple[callable, Tuple, Dict]]
2368:        callbacks : List[Tuple[callable, Tuple, Dict]]

lib/matplotlib/axes/_base.py
1613:        v : List[float] or one of the strings listed below.
```

**Comment by anntzer** [comment]
Added test.

File: lib/matplotlib/axes/_base.py (hunk 1 above)
Comment (timhoffm) [reply lib/matplotlib/axes/_base.py]: I'll accept this for now, even tough I would like to have it discussed and agreed upon. I'm not sure if this is really the best way. Whatever the policy is, it needs documentation. I'll leave the topic for #10225.

File: lib/matplotlib/artist.py [hunk 2]
```diff
@@ -1135,6 +1134,14 @@ def get_valid_values(self, attr):
         match = self._get_valid_values_regex.search(docstring)
         if match is not None:
             return re.sub("\n *", " ", match.group(1))
+
+        # Much faster than list(inspect.signature(func).parameters)[1],
+        # although barely relevant wrt. matplotlib's total import time.
+        param_name = func.__code__.co_varnames[1]
+        match = re.search("(?m)^ *{} : (.+)".format(param_name), docstring)
```
Comment (timhoffm) [inline lib/matplotlib/artist.py]: I'm not quite sure, what we want the exact sematics to be.

What about `foo : int, optional, default: None`?

Do we want just `int` or `int, optional, default: None`?

File: lib/matplotlib/artist.py [hunk 3]
```diff
@@ -1054,17 +1054,16 @@ def mouseover(self, val):
...
+
     def __init__(self, o):
         """
-        Initialize the artist inspector with an
-        :class:`~matplotlib.artist.Artist` or iterable of :class:`Artists`.
-        If an iterable is used, we assume it is a homogeneous sequence (all
-        :class:`Artists` are of the same type) and it is your responsibility
-        to make sure this is so.
+        Initialize the artist inspector with an `Artist` or an iterable of
+        `Artist`\s.  If an iterable is used, we assume it is a homogeneous
```
Comment (jklymak) [inline lib/matplotlib/artist.py]: This make the tests fail for me:   `SyntaxError: invalid escape sequence \s`  Do I have an old version of something lying around?

Comment (timhoffm) [reply lib/matplotlib/artist.py]: Handled by #11336

**Pull request merged**